File: api/src/main/java/io/bunnies/baas/services/v2/responses/BunnyResponseV2.java
package io.bunnies.baas.services.v2.responses;

import com.google.common.collect.Maps;
import io.bunnies.baas.resources.IBunnyResource;

import java.util.Map;

public class BunnyResponseV2 {
    public String id;
    public Map<String, String> media;
    public String source;
    public int thisServed;
    public int totalServed;

    public BunnyResponseV2(IBunnyResource bunnyResource, Iterable<String> fileTypeKeys, int thisServed, int totalServed) {
        this.id = bunnyResource.getBunnyID();
        this.media = Maps.newHashMap();
        this.source = bunnyResource.getBunnySource();
        this.totalServed = totalServed;
        this.thisServed = thisServed;

        for (String fileType : fileTypeKeys) {
            this.addMediaType(bunnyResource, fileType);
        }
    }

    public void addMediaType(IBunnyResource bunnyResource, String type) {
        if (bunnyResource.hasResourceType(type)) {
            this.media.put(type.toLowerCase(), bunnyResource.getResourceUrl(type));
        }
    }

    public int numberOfContainedFileTypes() {
        if (this.media == null) {
            return 0;
        }

        return this.media.keySet().size();
    }
}

File: api/src/main/java/io/bunnies/baas/services/v2/responses/BunnyResponseV2.kt
package io.bunnies.baas.services.v2.responses

import com.google.common.collect.Maps
import io.bunnies.baas.resources.IBunnyResource

class BunnyResponseV2(bunnyResource: IBunnyResource, fileTypeKeys: Iterable<String>, var thisServed: Int, var totalServed: Int) {
    val id: String
    val media = mutableMapOf<String, String>()
    val source: String

    init {
        this.id = bunnyResource.bunnyID
        this.source = bunnyResource.bunnySource

        for (fileType in fileTypeKeys) {
            this.addMediaType(bunnyResource, fileType)
        }
    }

    fun addMediaType(bunnyResource: IBunnyResource, type: String) {
        val resourceUrl = bunnyResource.getResourceUrl(type)
        if (resourceUrl != null) {
            media[type.toLowerCase()] = resourceUrl
        }
    }

    fun numberOfContainedFileTypes(): Int {
        return media.keys.size
    }
}
